### PR TITLE
Enhance extract_datetime(), add time utilities

### PR DESCRIPTION
**Pull request by penrods**
Many cases that were missed in the unittests for extract_datetime()
from the original source.  Restored those tests and made code
adjustments to support them all.

Also adding the mycroft.util.time module.  This supports:
* mycroft.util.time.default_timezone()
  Returns the user-configured timezone based on location
* mycroft.util.time.now_utc()
  Returns the time in UTC
* mycroft.util.time.now_local()
  Returns the time in the user's timezone
* mycroft.util.time.to_utc()
  Converts to UTC
* mycroft.util.time.to_local()
  Converts to user's timezone

NOTE: Several skills should be updated to use these now.

==== Fixed Issues ====
Several issues for skills regarding parsing of "today"

====  Documentation Notes ====
Note the new module:  mycroft.util.time

==== Localization Notes ====
Localized versions of extract_datetime() likely need to be
updated, as most were based on the original English implementation

## How to test
Run the test/unittests/util/test_parse.py unittest

## Contributor license agreement signed?
CLA [ X ] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

**Comment by forslund** [comment]
Looks pretty good! Having proper timezone info will make lots of things easier!

I'm missing the parsing of "now", which would be useful for cases like "do I have anything scheduled in an hour", "do I have anything scheduled right now".

I've found a small issue, not sure it's worth going after. (and it might very well be my input that's missing something)

when using a weekday and date like Friday the 20th it seem to regard `the 20th` as a time and not a date.

```python
>>> extract_datetime('asdf Friday the 20th')
[datetime.datetime(2018, 7, 13, 20, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Stockholm')), 'asdf']
>>> extract_datetime('asdf friday the 20')
[datetime.datetime(2018, 7, 13, 20, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Stockholm')), 'asdf']
>>> extract_datetime('asdf friday the twenty')
[datetime.datetime(2018, 7, 13, 0, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Stockholm')), 'asdf twenty']
>>> extract_datetime('asdf friday the twentieth')
[datetime.datetime(2018, 7, 13, 0, 0, tzinfo=tzfile('/usr/share/zoneinfo/Europe/Stockholm')), 'asdf twentieth']
```

File: mycroft/util/time.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    Args:
+        dt (datetime): A datetime (presumably in some local zone)
+    Returns:
+        (datetime): time converted to UTC
+    """
+    tzUTC = gettz("UTC")
+    if dt.tzinfo:
+        return dt.astimezone(tzUTC)
+    else:
+        return dt.replace(tzinfo=gettz("UTC")).astimezone(tzUTC)
```
Comment (forslund) [inline mycroft/util/time.py]: Isn't it enough to just do the 

`dt.replace(tzinfo=gettz("UTC"))`

The `astimezone()` would convert from UTC to UTC right?

Comment (forslund) [reply mycroft/util/time.py]: I understand, I just thought the `astimezone` was redundant in this case (I only did a quick test so I might be wrong) but I trust you that it's necessary.

Comment (penrods) [reply mycroft/util/time.py]: Yeah, we could special-case to skip calling astimezone() in the UTC -> UTC case, but astimezone() already has the code to handle it.

**Comment by forslund** [comment]
Merged through #1693, Closing

**Pull request closed without merging**